src/Symfony/Bundle/SecurityBundle/Resources/views/Collector/security.html.twig

```diff
@@ -41,6 +41,11 @@
                             <span class="sf-toolbar-status sf-toolbar-status-{{ is_authenticated ? 'green' : 'yellow' }}">{{ is_authenticated ? 'Yes' : 'No' }}</span>
                         </div>
 
+                        <div class="sf-toolbar-info-piece">
+                            <b>Roles</b>
+                            <span>{{ collector.roles|join(', ') }}</span>
```

Comment:
Should we limit the output here? I'm afraid we break the toolbar for apps with many roles. 🙈

Reply:
was checking :grimacing: i have 200+ roles :}

Reply:
😱

Reply:
yes. It's a symptom of permissions-as-roles. In that sense ive no idea what a reasonable no. of roles is, nor about an average role name length. I do agree the current toolbar has limited UI to display many data.

Reply:
Yeah, good point - limiting would be a good idea. Do you have any preference about this. I would see the following options:

- showing the first m roles, and something like "and n more"
- having something like a modal that could be opened if there are more than m roles

Reply:
Perhaps `N roles(s)` is sufficient?  With eg. a title attribute for the actual role names ... 

(ignore my 200+ roles, it's broken design)

Reply:
IMO: If this should be a good addition, as requested in #42763, this should not only contain the number of assigned roles

Reply:
We should agree on a threshold above which we just display the number of roles. If a user has 20 roles and we only display three arbitrary ones, I don't think the interface is helpful. If the number of roles is below the threshold, we simply list them.

Reply:
@derrabus do you have any specific number in mind?

Reply:
Dunno, five? 🤷🏻‍♂️

Reply:
I think one should be enough, if they want to know more detail about current roles just tab in the profiler and take a look.

Hard to imagine how does it look like when rendering lots of roles in a small UI.

Reply:
> this should not only contain the number of assigned roles

hence i suggested to use a title attribute and avoid this discussion :)

Reply:
I like @ro0NL's suggestion of using the title attribute:

![image](https://user-images.githubusercontent.com/749025/132107868-5fa34b91-c89f-4ea0-ae1b-66bcde86cdf9.png)

I think 1 role is a good threshold value :smile: . In this example screenshot, the user only has 4 roles and it already wouldn't fit the normal modal.
Also, I always recommend people to limit the number of roles a user has as close to 1 as possible. Maybe complex applications need 3+ roles, but generally you should be able to manage 1 or 2 roles if you split roles from permissions.

Reply:
i agree the dashed underline is likely enough to trigger a user hover, and go from there.

i'd opt out from a random number to display roles for

we could spare out "Logged in as" in favor of the "user icon" as in the toolbar itself

we could spare out "Authenticated: yes" in favor of the negative edge case (eg. "Not authenticated" as a single badge)

"Action" looks redundant :) could be some "logout icon" too

last but not least, IIUC the title attribute is limited to 512 chars. Perhaps we should invent some CSS only tooltip util.

Reply:
Please keep this discussion focussed and discuss cleaning up other things in another issue/pr :)

I'm 100% in favor of keeping this as simple as possible. Let's not increase difficulty with CSS/js tooltips. The longest role I've seen is Symfony's own `ROLE_PREVIOUS_ADMIN`, which is 19 characters long (meaning a user can have 24 roles before the title attribute overflows).

Reply:
Right, only Javier should do overhauls. I never implied "JS". We have a hard limit on title attributes.